app/services/qa/linked_data/mapper/search_results_mapper_service.rb

```diff
@@ -38,17 +38,31 @@ def map_values(graph:, predicate_map:, sort_key:, preferred_language: nil, conte
             graph.subjects.each do |subject|
               next if subject.anonymous? # skip blank nodes
               values = graph_mapper_service.map_values(graph: graph, predicate_map: predicate_map, subject_uri: subject) do |value_map|
-                next value_map if context_map.blank?
-                context = {}
-                context = context_mapper_service.map_context(graph: graph, context_map: context_map, subject_uri: subject) if context_map.present?
-                value_map[:context] = context
-                value_map
+                map_context(graph, sort_key, context_map, value_map, subject)
               end
-              search_matches << values unless sort_key.present? && values[sort_key].blank?
+              search_matches << values if result_subject? values, sort_key
```

Comment:
DRY:  Extracted to a private method so the check for sort_key can happen here and in the context mapping code.